core/src/test/java/org/apache/struts2/interceptor/FetchMetadataInterceptorTest.java

```diff
@@ -102,22 +107,179 @@ public void testPathInExemptedPaths() throws Exception {
+        mai.setInvocationContext(context);
+
+        cookieInterecptor.intercept(mai);
+        interceptor.intercept(mai);
+
+        assertEquals("Expected interceptor to NOT accept this request", "403", interceptor.intercept(mai));
+    }
+
+    public void testIntegrationRejectedRequestDiffOrder() throws Exception{
+        MockActionInvocation mai = new MockActionInvocation();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        CookieInterceptor cookieInterecptor = new CookieInterceptor();
+
+        request.addHeader("sec-fetch-site", "foo");
+
+        ServletActionContext.setRequest(request);
+        ServletActionContext.setResponse(response);
+        ActionContext context = ServletActionContext.getActionContext();
+        mai.setInvocationContext(context);
+
+        interceptor.intercept(mai);
+        cookieInterecptor.intercept(mai);
+
+        assertEquals("Expected interceptor to NOT accept this request", "403", interceptor.intercept(mai));
+    }
+
+    public void testIntegrationMultipleValidRequest() throws Exception{
+        MockActionInvocation mai = new MockActionInvocation();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        CookieInterceptor cookieInterecptor = new CookieInterceptor();
+        NoOpInterceptor noOpInterceptor = new NoOpInterceptor();
+
+        request.addHeader("sec-fetch-site", "foo");
+
+        ServletActionContext.setRequest(request);
+        ServletActionContext.setResponse(response);
+        ActionContext context = ServletActionContext.getActionContext();
+        mai.setInvocationContext(context);
+
+        cookieInterecptor.intercept(mai);
+        interceptor.intercept(mai);
+        noOpInterceptor.intercept(mai);
+
+        assertEquals("Expected interceptor to NOT accept this request", "403", interceptor.intercept(mai));
+    }
+
+    public void testIntegrationMultipleRejectedRequest() throws Exception{
+        MockActionInvocation mai = new MockActionInvocation();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        CookieInterceptor cookieInterecptor = new CookieInterceptor();
+        NoOpInterceptor noOpInterceptor = new NoOpInterceptor();
+
+        request.addHeader("sec-fetch-site", "foo");
+
+        ServletActionContext.setRequest(request);
+        ServletActionContext.setResponse(response);
+        ActionContext context = ServletActionContext.getActionContext();
+        mai.setInvocationContext(context);
+
+        cookieInterecptor.intercept(mai);
+        interceptor.intercept(mai);
+        noOpInterceptor.intercept(mai);
+
+        assertEquals("Expected interceptor to NOT accept this request", "403", interceptor.intercept(mai));
+    }
```

Comment:
These tests aren't really testing a chain of interceptors that gets interrupted. To be fair, we rely heavily on this behaviour just like all other interceptors, so feel free to either remove these tests. Sorry about that.